packages/clay-sticker/src/ClaySticker.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+import ClayIcon from '@clayui/icon';
+
+type DisplayType =
+	| 'danger'
+	| 'dark'
+	| 'info'
+	| 'light'
+	| 'primary'
+	| 'secondary'
+	| 'success'
+	| 'unstyled'
+	| 'warning';
+
+type Shape = 'circle' | 'rounded';
+
+type Position = 'bottom-left' | 'bottom-right' | 'top-left' | 'top-right';
+
+type Size = 'lg' | 'sm' | 'xl';
+
+interface Props extends React.HTMLAttributes<HTMLSpanElement> {
+	/**
+	 * Determines the color of the sticker.
+	 */
+	displayType?: DisplayType;
+
+	/**
+	 * Render ClayIcon in the ClaySticker element.
+	 */
+	icon?: string;
+
+	/**
+	 * Flag to indicate if the sticker should be positioned on the outside
+	 * corners when position is defined.
+	 */
+	outside?: boolean;
+
+	/**
+	 * Position of the sticker in relation to the contents.
+	 */
+	position?: Position;
+
+	/**
+	 * Shape of the sticker.
+	 */
+	shape?: Shape;
+
+	/**
+	 * Sticker size.
+	 */
+	size?: Size;
+
+	/**
+	 * The path to the SVG spritemap file containing the icons.
+	 */
+	spritemap?: string;
+}
+
+const ClaySticker: React.FunctionComponent<Props> = ({
+	children,
+	className,
+	displayType = 'primary',
+	icon,
+	outside = false,
+	position,
+	shape = 'rounded',
+	size,
+	spritemap,
+	...otherProps
+}) => (
+	<span
+		{...otherProps}
+		className={classNames('sticker', className, {
+			[`sticker-${shape}`]: shape,
+			[`sticker-${displayType}`]: displayType,
+			[`sticker-${position}`]: position,
+			[`sticker-outside`]: position && outside,
+			[`sticker-${size}`]: size,
+		})}
+	>
+		<span
+			className={classNames({
+				'inline-item': icon,
+				'sticker-overlay': !icon,
+			})}
+		>
+			{icon ? <ClayIcon spritemap={spritemap} symbol={icon} /> : children}
```

Comment:
How do you feel about de-coupling Icon from sticker and make is so that it'll be composed together? My thinking is that for cases where the Icon is the primary content, let the user compose that, however if the icon is integral to the functionality(like the 'X' on ClayLabel) then we can make it a dependency.

So things generic items like Button, Badge, Sticker, etc would all be independent of Icon and if a user wants to use Icon inside of that component, they can compose it together.

Reply:
Hey @bryceosterhaus I thought a lot about this before submitting, one of the deadlock it has been was that I need to replace the `sticker-overlay` class with `inline-item` I could add an API to create this condition but I think it would be unnecessary , because its use case would only be for `Icon` (I'm still creating a dependency of Icon within the component) and maybe in that way it can create a degree of dependency to make the composition with `Icon` work fine (We can decrease this adding in the documentation as a guide to good practices...).

But I really like to remove the dependencies and make it more flexible, but I still do not know how we could dribble this case here.

Reply:
What if we checked the contents of `children` and conditionally used `inline-item` if it was an svg?

Reply:
We would have to go through the complete Fiber tree to see if this component exists, I think it's not worth it and it's very costly depending on the size of the application. We could check the **props** of **children** if there is a `symbol` but I find this check too weak.

I was thinking of checking the `type` of the **children** to know the name of the component but the` type.name` is according to the name of the function and that name may disappear when minified and receive optimizations for production.

Reply:
I talked with @pat270 about it and he decided that the behavior is weird on the css side to require `inline-item` for icons in a sticker. Take a look at https://github.com/liferay/clay/issues/1825. I think we can go ahead with the composition route with this new issue resolving the `inline-item` issue.

Reply:
Perfect, I will update with these changes.